mlir/lib/Dialect/SPIRV/IR/SPIRVTypes.cpp

```diff
@@ -514,6 +514,12 @@ bool ScalarType::isValid(IntegerType type) {
 
 void ScalarType::getExtensions(SPIRVType::ExtensionArrayRefVector &extensions,
                                std::optional<StorageClass> storage) {
+  if (llvm::isa<BFloat16Type>(*this)) {
+    static const Extension exts[] = {Extension::SPV_KHR_bfloat16};
+    ArrayRef<Extension> ref(exts, std::size(exts));
+    extensions.push_back(ref);
```

Comment:
```suggestion
    static const auto ext = Extension::SPV_KHR_bfloat16;
    extensions.push_back(ext);
```
ArrayRef has a constructor that takes a single element